src/main/java/edu/mit/puzzle/cube/core/model/HuntStatusStore.java

```diff
@@ -327,6 +329,44 @@ public boolean setTeamProperty(
         return updates > 0;
     }
 
+    public <P extends Team.Property> boolean mutateTeamProperty(
```

Comment:
Excellent. The mutation of the property is actually happening while the database transaction is open (that's why the caller needs to pass in a Function to do the mutation, instead of doing the read themselves followed by a mutation followed by an update). So with the isolation level now set to SERIALIZABLE, I think we're good :-)